pkg/frontend/querymiddleware/model_extra.go

```diff
@@ -42,63 +43,232 @@ func newEmptyPrometheusResponse() *PrometheusResponse {
 	}
 }
 
+type PrometheusRangeQueryRequest struct {
```

Comment:
Do we need all the `protobuf` tags still? (same thing applies to other structs below)

And do we still need the `json` tags?